### PR TITLE
Feat/debugging write transactions

### PR DESCRIPTION
**Pull request by okdistribute**
This is a key thing we need to understand and support customers with

### CODE REVIEW

**Review by connorpower — APPROVED** [review]
Somewhat intimidating to see my internal support doc hoisted straight to the website 😅 But it's great to start getting this self-help information in front of customers! Thanks for doing it.

I've refined a bunch of the text for public eyes, and improved some wording where I could.

File: docs/common/troubleshooting.mdx [hunk 1]
```diff
@@ -100,10 +105,219 @@ your IT administrator to see if your computer is configured to block WebSocket c
...
+This can happen when the device has received large amounts of data from multiple peers **simultaneously** and must update the database with the changes it received from each. Ditto can only update the database with sync data from remote peers one at a time, so the other updates must wait their turn.
+
+This scenario is most likely to happen during a large initial sync - either with the Big Peer **at the same time** as with nearby devices, or just with **multiple** nearby devices when joining a mesh for the first time (or re-joining after an extended period away).
+
+This scenario shouldn’t necessarily be a problem for the user unless it winds up blocking a `User` transaction.
+
+Future work is planned do:
+
+1. Allow user writes to take precedence and interrupt replication. This will allow an app to stay responsive to user edits.
+2. Have replication begin in a “warm-up” phase where we limit the number of simultaneous  peers we sync with until we’ve progressed past the initial sync phase.
```
[the remark this answers is not in the document]
Comment (okdistribute) [reply docs/common/troubleshooting.mdx:294]: Agree!

File: docs/common/troubleshooting.mdx [hunk 2]
```diff
@@ -100,10 +105,219 @@ your IT administrator to see if your computer is configured to block WebSocket c
...
+From the application code, it might not be obvious what the problem is. By looking in the logs, you might get a hint. For instance:
+
+`LOG_LEVEL: Waiting for write transaction (elapsed: XXs), originator=User blocked_by=Replication`
+
+Here we can see that replication is blocking our user’s write transaction. This might happen if we’ve just received a large amount of sync data from another peer. Most commonly, happens during initial sync (either with the Big Peer, or joining a mesh for the first time, or re-joining a mesh after an extended period away).
+
+This scenario is something to be aware of, but there’s no solution yet. The user transaction will eventually unblock and continue when replication has finished updating the store. Future work is planned to:
+
+1. Allow user writes to take precedence and interrupt replication. This will allow an app UI to continue feeling responsive to user edits.
+2. Observable replication. This will allow an app to display a loading screen during initial sync to ensure the user can’t interact with an app before its initial data is ready.
```
[the remark this answers is not in the document]
Comment (okdistribute) [reply docs/common/troubleshooting.mdx:277]: Agree!

**Comment by okdistribute** [comment]
I know it's nerve-wracking!! But you're a great writer and produce amazing documents that can save developers a ton of time. I believe we should start being somewhat transparent, especially with debugging, because we'd want to share this information with customers anyway. Thanks for spending time to clarify and update parts that had language that was a bit too internal.

**Pull request merged**